compiler/src/dotty/tools/dotc/core/ContextOps.scala

```diff
@@ -59,13 +59,16 @@ object ContextOps:
         val preSym = pre.typeSymbol
         // 1. Try to search in current type and parents.
         val directSearch =
-          if name.isTypeName && name.endsWith(StdNames.str.MODULE_SUFFIX) then
-            pre.findMember(name.stripModuleClassSuffix, pre, required, excluded) match
-              case NoDenotation => NoDenotation
-              case symDenot: SymDenotation =>
-                symDenot.companionModule.denot
-          else
-            pre.findMember(name, pre, required, excluded)
+          def asModule =
+            if name.isTypeName && name.endsWith(StdNames.str.MODULE_SUFFIX) then
+              pre.findMember(name.stripModuleClassSuffix, pre, required, excluded) match
```

Comment:
I suggest to convert the name to a `moduleClassName` (after stopping the suffix), in that case you should hopefully find the real module class in scope
```suggestion
              pre.findMember(name.stripModuleClassSuffix.moduleClassName, pre, required, excluded) match
```